compiler/rustc_borrowck/src/places_conflict.rs

```diff
@@ -211,7 +211,7 @@ fn place_components_conflict<'tcx>(
             let proj_base = &borrow_place.projection[..access_place.projection.len() + i];
             let base_ty = Place::ty_from(borrow_local, proj_base, body, tcx).ty;
 
-            match (elem, &base_ty.kind(), access) {
+            match (elem, &base_ty.strip_variant_type().kind(), access) {
```

Comment:
We should document each such use to explain why it doesn't matter. E.g. this case is because we only care about whether the type has a drop impl itself. Though we could just add a method to Ty that returns whether it has a drop impl and make that function work for both variant and adt.

Reply:
I went with the adding a method to Ty approach, seemed better than using strip_variant_type here.